### PR TITLE
Added valkey module

### PR DESCRIPTION
**Pull request by JensvandeWiel**
## What does this PR do?
It adds a dedicated module For [valkey](https://github.com/valkey-io/valkey) using the Redis module as base.

## Why is it important?
Valkey can work with the Redis module, but since it is not the same project and since it can change compared to Redis it should have it's own module.

### CODE REVIEW

**Comment by mdelapenya** [comment]
Hi @JensvandeWiel thanks for submitting the new module. Do you think you can extract the valkey module changeset to a separate PR? There are changes to the go modules that should not apply to the module creation. Or you think the changes are related? If not, I'd suggest sending two PRs: one for the module, one for the dependency updates

**Comment by mdelapenya** [comment]
> Hey, I thought I needed to update dependencies as it says in the contribution guide. I can probably remove the dependencies updates from the other modules if you want

Yes please, probably the contribution guides are misleading, talking about the core, not the modules.

In any case, I'd prefer one PR for the valkey module, and one for the deps 🙏

**Comment by JensvandeWiel** [comment]
Ok, ill do that later today

**Comment by JensvandeWiel** [comment]
Ok I've removed the other changes, I found that I still need to change the client to use the valkey client, so ill do that first, so please wait with merging.

**Comment by mdelapenya** [comment]
Cool thank you @JensvandeWiel! Did you read the module contribution guide [here](https://golang.testcontainers.org/modules/#creating-a-new-module)? It seems the code generation tool was not used, so the PR just includes the module code, but nothing about the docs, and the GH action, etc. Could you please run the generator adding your code on top of the generated scaffolding? 🙏

**Comment by JensvandeWiel** [comment]
Oh, I'm sorry ill do it soon!

**Comment by JensvandeWiel** [comment]
Ok, I've added it!

File: modules/valkey/valkey_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	for i := 0; i < keyCount; i++ {
+		// Set data
+		key := fmt.Sprintf("{user.%s}.favoritefood.%d", uuid.NewString(), i)
+		value := fmt.Sprintf("Cabbage Biscuits %d", i)
+
+		ttl, _ := time.ParseDuration("2h")
+
+		err = client.Do(ctx, client.B().Set().Key(key).Value(value).Exat(time.Now().Add(ttl)).Build()).Error()
```
Comment (mdelapenya) [inline modules/valkey/valkey_test.go:139]: Handle error here
```suggestion
		err = client.Do(ctx, client.B().Set().Key(key).Value(value).Exat(time.Now().Add(ttl)).Build()).Error()
		require.NoError(t, err)

```

File: modules/valkey/valkey_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	"fmt"
+	tcvalkey "github.com/testcontainers/testcontainers-go/modules/valkey"
+	"github.com/valkey-io/valkey-go"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
```
Comment (mdelapenya) [inline modules/valkey/valkey_test.go:14]: Proper imports format ( I ran `make lint` from the module)
```suggestion
import (
	"context"
	"fmt"
	"path/filepath"
	"testing"
	"time"

	"github.com/google/uuid"
	"github.com/stretchr/testify/require"
	"github.com/valkey-io/valkey-go"

	tcvalkey "github.com/testcontainers/testcontainers-go/modules/valkey"
)
```

**Review by mdelapenya — COMMENTED** [review]
@JensvandeWiel I left a few comments regarding linting. Could you address them?

Other than that, LGTM thank you so much for your contribution! 🚀 

I'll merge once the comments are addressed and the CI passes (it already pass locally, so I'm confident it will).

Cheers!

**Comment by JensvandeWiel** [comment]
Ok, I've added the changes and updated it with main branch.

**Comment by mdelapenya** [comment]
The render URL looks pretty nice! https://deploy-preview-2639--testcontainers-go.netlify.app/modules/valkey/#container-methods well done!

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Comment by mdelapenya** [comment]
Module is live in the modules catalog: https://github.com/testcontainers/community-module-registry/pull/62

**Pull request merged**